### ISSUE TITLE
Extract feature layer 

### ISSUE THREAD

**Issue by nerddd** [happynear/NormFace#15]
I used your scaled_cosine_softmax.prototxt，I don't know which layer do extract feature when applying,is 'last_ip_layer' ? 'normalize1' ? or something else? Thanks

**Comment by liuruqin001** [comment]
last_ip_layer is  net  last  layer name。

sample :

name: "Face-ResNet"
...
layer {
  name: "fc5"
  type: "InnerProduct"
  bottom: "res5_6"
  top: "fc5"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 512
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}

modify :  

############################## modify

name: "Face-ResNet"
...

layer {
  name: "normalize1"
  type: "Normalize"
#  "last_ip_layer"  modify  to fc5
  bottom: "fc5"  
  top: "norm1"
}
layer {
  name: "id_weight_l2"
  type: "Parameter"
  top: "id_weight_l2"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  parameter_param {
    shape {
      dim: 19469
      dim: 512
    }
    blob_filler {
      type: "gaussian_unitball"
    }
  }
}

......

**Comment by nerddd** [comment]
@liuruqin001 Thanks for your answer.But I think I'm misunderstood. I've already trained a model use the normface, but  sorry , my question is :when testing ,extract feature,which layer should I choose? It seems to be "fc5"?

**Comment by happynear** [comment]
Yes, fc5. Don't forget to use the mirror face trick https://github.com/happynear/NormFace/blob/master/prototxt/example_of_mirror_face.prototxt .

**Comment by nerddd** [comment]
@happynear Thanks, I've done the experiment. But the result on lfw doesn't improved. I used center loss before,and got 99.37%. This time, when I trained normface, I found there was something wrong when I put a center loss layer. So I only use a softmax loss layer in normface, and got 99.37% too. Could you please show me how to put  center loss in normface?Maybe it can help. By the way, mirror is really a good trick.

**Comment by happynear** [comment]
There is no need to add center loss layer anymore after normalization. I guess maybe you have used a too small learning rate when fine-tuning the normalized softmax loss and maybe you need to train more iterations.

**Comment by nerddd** [comment]
No need? Ok, thanks, I'll have a another try
